scanpy/get.py

```diff
@@ -166,9 +168,17 @@ def obs_df(
     var_names = []
     var_symbol = []
     not_found = []
-    for key in keys:
+
+    # use only unique keys, otherwise duplicated keys will
+    # further duplicate when reordering the keys later in the function
+    for key in np.unique(keys):
         if key in adata.obs.columns:
             obs_names.append(key)
+            if key in gene_names.index:
+                logg.warning(
+                    f'The key `{key}` is found in both adata.obs and adata.var_names.'
+                    'Only the adata.obs key will be used.'
+                )
```

Comment:
Use actual warnings, they are easier for users to deal with.

Reply:
done